tools/setup_helpers/cuda.py

```diff
@@ -4,27 +4,19 @@
 import ctypes.util
 from subprocess import Popen, PIPE
 
+from . import which
 from .env import IS_WINDOWS, IS_LINUX, IS_DARWIN, check_env_flag, check_negative_env_flag
 
 LINUX_HOME = '/usr/local/cuda'
 WINDOWS_HOME = glob.glob('C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v*.*')
 
 
 def find_nvcc():
-    if IS_WINDOWS:
-        proc = Popen(['where', 'nvcc.exe'], stdout=PIPE, stderr=PIPE)
-    else:
-        proc = Popen(['which', 'nvcc'], stdout=PIPE, stderr=PIPE)
-    out, err = proc.communicate()
-    out = out.decode().strip()
-    if len(out) > 0:
+    nvcc = which('nvcc')
+    if nvcc is not None:
         if IS_WINDOWS:
-            if out.find('\r\n') != -1:
-                out = out.split('\r\n')[0]
-            out = os.path.abspath(os.path.join(os.path.dirname(out), ".."))
-            out = out.replace('\\', '/')
-            out = str(out)
-        return os.path.dirname(out)
+            nvcc = nvcc.replace('\\', '/')
```

Comment:
You can use the `escape_path` utility here

Reply:
Done!

Reply:
I should have added that since you use `escape_path`, the `if IS_WINDOWS:` line is no longer necessary (:

Reply:
No problem, fixed.